advanced/Templates/pihole-FTL.service.ini

```
@@ -0,0 +1,37 @@
+[Unit]
+Description=pihole-FTL Domain Name Server
+Documentation=man:pihole-FTL(8)
+After=remote-fs.target
+
+# Alternative based on https://wiki.debian.org/Hardening/Daemons#bind
+#   allows other services which rely on DNS to use pihole-FTL, but means
+#   breaking if config relies on remote-fs
+# After=network.target
+# Wants=nss-lookup.target
+# Before=nss-lookup.target
+
+After=resolvconf.service
+Wants=resolvconf.service
+# fail since dnsmasq is embedded
+Conflicts=dnsmasq.service
+
+[Service]
+# Run as unprivileged user, use AmbientCapabilities to assign capabilities
+User=pihole
+Group=pihole
+
+# start()
+ExecStartPre=+/bin/sh /etc/pihole/fix_files.sh
+ExecStartPre=+/bin/sh -c "echo \"nameserver 127.0.0.1\" | /sbin/resolvconf -a lo.piholeFTL"
+# Assign capabilities to the pihole user slice
+AmbientCapabilities=CAP_NET_BIND_SERVICE CAP_NET_RAW CAP_NET_ADMIN
+ExecStart=/usr/bin/pihole-FTL no-daemon
+
```

Previously, I proposed
```suggestion
; Make /usr, /boot, /etc and possibly some more folders read-only...
ProtectSystem=full
; ... except /etc/pihole
; This merely retains r/w access rights, it does not add any new.
; Must still be writable on the host!
ReadWriteDirectories=/etc/pihole
```
Do you think this is a good idea?

---

It looks like this might break existing setups if people changed their configs: 
[Configurable paths](https://github.com/pi-hole/FTL/blob/9e8273b748a6a0870c64fcc0ec4880a28bbcd587/config.c#L282-L310)

It might be interesting as an optional security feature which could be installed as a drop-in config like `/etc/systemd/system/pihole-FTL.service.d/protect-fs.conf`

---

Ah yes, I basically forgot about the possibility to move all the files. In this case, we probably shouldn't do it.